src/monodroid/CMakeLists.txt

```diff
@@ -289,6 +292,14 @@ endif()
 
 set(XAMARIN_APP_STUB_SOURCES ${SOURCES_DIR}/application_dso_stub.cc)
 add_library(xamarin-app SHARED ${XAMARIN_APP_STUB_SOURCES})
+if (ENABLE_NDK)
+  # Only Android builds need to go in separate directories, desktop builds have the same ABI
```

Comment:
I don't *immediately* see where/why the ABI differs, but why must the ABI differ in the first place?  Is this strictly necessary?

Reply:
Yes, because `Release` builds use `[MVID][32-bit-int]` for map storage, mixed with `[JavaName][32-bit-int]` and `Debug`, with this PR, uses `[TypeName][32-bit-int]` for indexing and stores the map differently. It also needs to fix up the pointers on load time in `Debug+InstantRun` case. Also, `Release` uses more public symbols in `libxamarin-app.so` than `Debug` (5 vs 1, respectively). Cramming all of this into a single structure would become confusing and messy, so I made the ABI differ. I *could* leave both `Release` and `Debug` structures and fields in all builds, but that's just... untidy.